src/format.rs

```diff
@@ -663,3 +805,17 @@ impl fmt::Display for Codec {
         }
     }
 }
+
+impl std::ops::Deref for CodecConfig {
```

Comment:
I think of it like this.

The codec configurations is essentially a list of payload (parameters). We could have had just a `Vec<PayloadParameter>`, but there are a couple of nice functions (`match` etc) that are part of the public API.

If I put it like that, does that make it any better?

Reply:
Somewhat, but I think it can still be confusing. `Deref` is a bit too magic, I'd prefer explicit methods instead

Reply:
How about I provide that as external API? Leave the deref. Doesn't make much difference if it's possible to Deref _as well as_?

Reply:
I'm taking my cues from the Deref docs:

> Implementing Deref for smart pointers makes accessing the data behind them convenient, which is why they implement Deref. On the other hand, the rules regarding Deref and [DerefMut](dfile:///Users/hugotunius/Library/Application%20Support/Dash/DocSets/Rust/Rust.docset/Contents/Resources/Documents/doc.rust-lang.org/1.71.1/std/ops/trait.DerefMut.html) were designed specifically to accommodate smart pointers. Because of this, **Deref should only be implemented for smart pointers** to avoid confusion.

Our use of Deref for newtypes I think is okayish, but it feels like a stretch here